src/main/java/uk/gov/pay/directdebit/payments/dao/mapper/GoCardlessEventMapper.java

```diff
@@ -31,14 +32,18 @@ public GoCardlessEvent map(ResultSet resultSet, StatementContext statementContex
                 .withLinksNewCustomerBankAccount(resultSet.getString("links_new_customer_bank_account"))
                 .withLinksOrganisation(resultSet.getString("links_organisation"))
                 .withLinksParentEvent(resultSet.getString("links_parent_event"))
-                .withLinksPayment(resultSet.getString("links_payment"))
                 .withLinksPayout(resultSet.getString("links_payout"))
                 .withLinksPreviousCustomerBankAccount(resultSet.getString("links_previous_customer_bank_account"))
                 .withLinksRefund(resultSet.getString("links_refund"))
                 .withLinksSubscription(resultSet.getString("links_subscription"))
                 .withCreatedAt(ZonedDateTime.ofInstant(
                         resultSet.getTimestamp("created_at").toInstant(), ZoneOffset.UTC));
 
+        String paymentId = resultSet.getString("links_payment");
+        if (paymentId != null) {
+            builder.withLinksPayment(GoCardlessPaymentId.valueOf(paymentId));
+        }
```

Comment:
The above could be done with `Optional.ofNullable(resultSet.getString("links_payment")).ifPresent...`

Reply:
🥇 great idea! We've updated them all.